acceptance/server_test.go

```diff
@@ -26,15 +38,108 @@ func NewTestServer() *TestServer {
 	return &TestServer{
 		Server: server,
 		Mux:    mux,
+		directories: map[string]bool{
+			"/Workspace": true,
+		},
+		files:     map[string][]byte{},
+		jobs:      map[int64]jobs.Job{},
+		nextJobId: 1,
+	}
+}
+
+func (s *TestServer) getStatus(path string) (workspace.ObjectInfo, error) {
+	if s.directories[path] {
+		return workspace.ObjectInfo{
+			ObjectType: "DIRECTORY",
+			Path:       path,
+		}, nil
+	} else if _, ok := s.files[path]; ok {
+		return workspace.ObjectInfo{
+			ObjectType: "FILE",
+			Path:       path,
+			Language:   "SCALA",
+		}, nil
+	} else {
+		return workspace.ObjectInfo{}, NotFoundError
 	}
 }
 
+func (s *TestServer) workspaceMkdirs(request workspace.Mkdirs) (string, error) {
+	s.directories[request.Path] = true
+
+	return "{}", nil
+}
+
+func (s *TestServer) workspaceImportFiles(path string, body []byte) (any, error) {
+	s.files[path] = body
+
+	return "{}", nil
+}
+
+func (s *TestServer) jobsCreate(request jobs.CreateJob) (jobs.CreateResponse, error) {
+	jobId := s.nextJobId
+	s.nextJobId++
+
+	jobSettings := jobs.JobSettings{}
+	err := jsonConvert(request, &jobSettings)
+	if err != nil {
+		return jobs.CreateResponse{}, fmt.Errorf("failed to convert request to job settings: %w", err)
+	}
+
+	s.jobs[jobId] = jobs.Job{
+		JobId:    jobId,
+		Settings: &jobSettings,
+	}
+
+	return jobs.CreateResponse{JobId: jobId}, nil
+}
+
+func (s *TestServer) jobsGet(jobId string) (jobs.Job, error) {
+	id := jobId
+
+	jobIdInt, err := strconv.ParseInt(id, 10, 64)
+	if err != nil {
+		return jobs.Job{}, fmt.Errorf("failed to parse job id: %w", err)
+	}
+
+	job, ok := s.jobs[jobIdInt]
+	if !ok {
+		return jobs.Job{}, NotFoundError
+	}
+
+	return job, nil
+}
+
+func (s *TestServer) jobsList() (jobs.ListJobsResponse, error) {
+	list := make([]jobs.BaseJob, 0, len(s.jobs))
+	for _, job := range s.jobs {
+		baseJob := jobs.BaseJob{}
+		err := jsonConvert(job, &baseJob)
+		if err != nil {
+			return jobs.ListJobsResponse{}, fmt.Errorf("failed to convert job to base job: %w", err)
+		}
+
+		list = append(list, baseJob)
+	}
+
+	return jobs.ListJobsResponse{
+		Jobs: list,
+	}, nil
+}
+
 func (s *TestServer) Handle(pattern string, handler HandlerFunc) {
 	s.Mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
 		resp, err := handler(r)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
+			if errors.Is(err, NotFoundError) {
+				payload := `{"error_code":"RESOURCE_DOES_NOT_EXIST","message":"Does not exist"}`
```

Comment:
Not sure this is the correct way. But we need to send the correct payload back, or SDK can't parse it.